File: LeetCode/1. Two Sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()
        for k, v in enumerate(nums):
            if target - v in d:
                return [d.get(target-v), k]
            else:
                d[v] = k


s = Solution()
print(s.twoSum([3, 2, 4], 6))

File: LeetCode/1. Two Sum/Solution.scala
import scala.collection.mutable

object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    val cache = new mutable.HashMap[Int, Int]
    for ((v, i) <- nums.zipWithIndex) {
      if (cache contains target - v) {
        return Array(cache(target - v), i)
      } else {
        cache(v) = i
      }
    }
    throw new IllegalArgumentException()
  }
}
